__tests__/screens/ChannelListScreenV2/ChannelListScreenV2.test.js

```diff
@@ -68,3 +144,50 @@ describe('ChannelListScreenV2', () => {
     expect(mockRefreshAnonymousChannelList).toHaveBeenCalled();
   });
 });
+
+jest.mock('react-native-push-notification', () => ({
+  checkPermissions: jest.fn()
+  // Add other methods from PushNotification that you use
+}));
+
+describe('ChannelListScreenV2 useEffect', () => {
+  let alertSpy;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    alertSpy = jest.spyOn(Alert, 'alert'); // Spy on Alert.alert
+  });
+
+  afterEach(() => {
+    alertSpy.mockRestore(); // Restore the original implementation after each test
+  });
+
+  it('should not show prompt if screen is not focused', async () => {
+    const mockIsFocused = jest.fn(() => false);
+    jest.mock('@react-navigation/native', () => ({
+      ...jest.requireActual('@react-navigation/native'),
+      useNavigation: () => ({
+        isFocused: mockIsFocused
+      })
+    }));
+
+    render(<ChannelListScreenV2 />, {wrapper: Store});
+    jest.advanceTimersByTime(20000); // Advance timers by 20 seconds
+    expect(PushNotification.checkPermissions).not.toHaveBeenCalled();
+  });
+
+  it('should handle authorizationStatus 1 with needsPermission', async () => {
+    PushNotification.checkPermissions.mockImplementation((callback) => {
+      callback({authorizationStatus: 1, alert: false, badge: true, sound: true});
+    });
+    StorageUtils.lastPromptNotification.get.mockReturnValue(jest.getRealSystemTime);
+
+    render(<ChannelListScreenV2 />, {wrapper: Store});
+    jest.advanceTimersByTime(20000);
+    expect(StorageUtils.lastPromptNotification.set).not.toHaveBeenCalled();
+    expect(Alert.alert).not.toBeCalled();
+  });
+
+  // Add similar tests for other cases and scenarios
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with the test case on lines 180-190. The `jest.advanceTimersByTime(20000);` function is used to simulate the passage of time, but it's not clear if there are any timers set up in the `ChannelListScreenV2` component that would be affected by this. If there aren't, this line won't have any effect.

Also, the mock implementation for `StorageUtils.lastPromptNotification.get` on line 184 returns `jest.getRealSystemTime`, which is a function, not a timestamp. This could cause issues when comparing the returned value to an actual timestamp. It should return a timestamp instead.

```diff
-    StorageUtils.lastPromptNotification.get.mockReturnValue(jest.getRealSystemTime);
+    StorageUtils.lastPromptNotification.get.mockReturnValue(Date.now());
```

Lastly, the test case on lines 180-190 checks if `StorageUtils.lastPromptNotification.set` and `Alert.alert` were not called, but it doesn't check if `PushNotification.checkPermissions` was called, which is the main function being tested here. You should add an assertion to check if `PushNotification.checkPermissions` was called.

```diff
+    expect(PushNotification.checkPermissions).toHaveBeenCalled();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->